confmap/resolver.go

```diff
@@ -34,11 +35,16 @@ type ResolverSettings struct {
 	// It is required to have at least one location.
 	URIs []string
 
-	// ProviderFactories is a list of Provider creation functions.
-	// It is required to have at least one ProviderFactory
-	// if a Provider is not given.
+	// ProviderFactories is a slice of Provider creation functions.
+	// It is required to have at least one provider.
 	ProviderFactories []ProviderFactory
 
+	// DefaultProvider is the provider that is used if ${} syntax is used but no schema is provided.
+	// If no DefaultProvider is set, ${} with no schema will not be expanded.
+	// It is strongly recommended to set an EnvProvider as the default to align with the
+	// OpenTelemetry Configuration Specification
+	DefaultProvider ProviderFactory
```

Comment:
Should we require the `DefaultProvider`  be in the `ProviderFactories` list? Should we automatically add `DefaultProvider` to `ProviderFactories` in `NewResolver`?

If we want to make the use of `DefaultProvider` as close to the current workflow as possible we need it to be in the providers map.

Reply:
But I also implemented a way that allows `DefaultProvider` to not be in the providers list.

Reply:
>But I also implemented a way that allows `DefaultProvider` to not be in the providers list.

I think that's alright, throwing an error is probably non-ideal if it's not in there. Could we add it to the `ProviderFactories` list in this case? I think it's weird to have a "default" that is not in the list of what you could use if you actually provided a scheme.

Reply:
If we agree that adding the defaultprovider to the providers map is ok, that will simplify other aspects of the feature

Reply:
Would probably accept a "string" and call this "DefualtSchema".